File: NERP.NetworkEntities/Models/GeoNode.cs
using System.Collections.Generic;

namespace NERP.NetworkEntities.Models
{
    public class GeoNode
    {
        public string Id { get; set; }
        public int Level { get; set; }
        public string LevelName { get; set; }
        public double[] Coordinates { get; set; }
        public double Connections { get; set; }
        public List<GeoNode> Children { get; set; }
        public List<Segment> Segments { get; set; }

    }

    public class Segment
    {
        public string Id { get; set; }
        public string Start { get; set; }
        public string End { get; set; }
        public string FullPath { get; set; }
        public List<List<double>> Coordinates { get; set; }

        public List<Accessory> Accessories { get; set; }
        public List<Civilwork> Civilworks { get; set; }
        public Cable Cable { get; set; }
        public List<Chamber> Chambers { get; set; }
        public Duct Ducts { get; set; }
        public List<ODF> Odfs { get; set; }
        public List<Pole> Poles { get; set; }
        public List<Splitter> Splitters { get; set; }
        public List<SpliceClosure> SpliceClosures { get; set; }
        public MicroDuct microduct { get; set; }
        public List<POP> pops { get; set; }
        public List<FAT> fats { get; set; }
        public List<FDT> fdts { get; set; }
    }
}

File: NERP.RND.Web/ClientApp/src/app/common/geo/geoNode.ts
import { Position, Point } from './geometry';
import { Cable } from '../../entities/models/cable';
import { Helper } from '../helper';
//import { Accessory } from '../../designs/models/design';
import { Civilwork } from '../../civilworks/models/civilwork';
import { Chamber } from '../../entities/models/chamber';
import { Pole } from '../../entities/models/pole';
import { Splitter } from '../../entities/models/splitter';
import { SpliceClosure } from '../../entities/models/spliceClosure';
import { Duct } from '../../entities/models/duct';
import { ODF } from '../../entities/models/odf';
import { Accessory } from '../../accessories/models/accessory';
import { Homepass } from '../../homepasses/models/homepass';
import { PoP } from '../../entities/models/pop';
import { MicroDuct } from '../../entities/models/MicroDuct';
import { FAT } from '../../entities/models/fat';
import { FDT } from '../../entities/models/Solution2.zip';

export class GeoNode {
    id?: string;
    levelName?: string;
    level: number;
    coordinates?: Position;
    connections?: number;
    children?: GeoNode[];
    segments?: Segment[];

    constructor(id?:string, coordinates?: Position) {

        //if (id == null || id == "")
        //    id = Helper.generateObjectId();

        this.coordinates = coordinates;
    }
}

export class Segment {
    id: string;
    start: string;
    end: string;
    coordinates: Array<Position>;
    accessories?: Accessory[];
    civilworks?: Civilwork[];
    cable?: Cable;
    chambers?: Chamber[];
    ducts?: Duct;
    odfs?: ODF[];
    poles?: Pole[];
    splitters?: Splitter[];
   spliceClosures?: SpliceClosure[];
    homepasses?: Homepass[];
    microduct?: MicroDuct;
    pops?: PoP[];
    fats?: FAT[];
    fdts?: FDT[]
}
